### ISSUE TITLE
Unpacking one element dict to typed dict castes value to all keys 

### ISSUE THREAD

**Issue by PerchunPak** [microsoft/pyright#5679]
**Describe the bug**

When we unpack a one-element dict to typed dict (even with `total=False`), pyright castes value to all keys. Hard to explain, so just look at the example.

**Code or Screenshots**

```py
import typing

class Something(typing.TypedDict):
    abc: str
    some: int
    other: int
    another: int

a = Something(**{"abc": "abc"})
```

```
/home/perchun/programming/pyright/to_test.py
  /home/perchun/programming/pyright/to_test.py:9:17 - error: Argument of type "str" cannot be assigned to parameter "some" of type "int" in function "__init__"
    "str" is incompatible with "int" (reportGeneralTypeIssues)
  /home/perchun/programming/pyright/to_test.py:9:17 - error: Argument of type "str" cannot be assigned to parameter "other" of type "int" in function "__init__"
    "str" is incompatible with "int" (reportGeneralTypeIssues)
  /home/perchun/programming/pyright/to_test.py:9:17 - error: Argument of type "str" cannot be assigned to parameter "another" of type "int" in function "__init__"
    "str" is incompatible with "int" (reportGeneralTypeIssues)
3 errors, 0 warnings, 0 informations
```

But if we will add one more element to dict (e.g. `{"abc": "abc", "some": 1}`), the errors will disappear.

---

With `git bisect`, found that this issue started to appear after d345e20ebf5fb1d9e322aa3560187571a7f8cf16 (1.1.312).

**Comment by erictraut** [comment]
This behavior is correct given the way TypedDicts work. Mypy generates the same error in this case.

There's no need to call the TypedDict's constructor if you already have a `dict` expression. You can simply use:
```python
class Something(typing.TypedDict, total=False):
    abc: str
    some: int
    other: int
    another: int

a: Something = {"abc": "abc"}
```

If, for some reason, you really want to construct the dictionary a second time (making a copy in the process), you would you need to do so as follows:
```python
a: Something = Something({"abc": "abc"})
```

Or the following works:
```python
a: Something = Something(abc="abc")
```

The reason that unpacking doesn't work here is because the expression `{"abc": "abc"}` (prior to unpacking) is evaluated as the type `dict[str, str]`, so all information about its constituent keys is lost prior to unpacking.
